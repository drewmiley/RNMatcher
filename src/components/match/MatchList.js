import React, { Component } from 'react';
import { View } from 'react-native';

import MatchView from './MatchView';

export default class MatchList extends Component {
    constructor(props) {
        super(props);
        this.shouldComponentUpdate = (nextProps, nextState) => {
            return this.props.matches !== nextProps.matches;
        };
    };
    render() {
        return <View>
            {this.props.matches.map(match =>
                <MatchView
                    key={match.get('id')}
                    matchPlayers={match.get('players')}
                    {...this.props} />
            )}
        </View>
    };
};