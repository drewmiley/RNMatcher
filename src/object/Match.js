import {List, Map} from 'immutable';

export default Player = (id, playerOne, playerTwo, immutable) => {
    const players = immutable ? List.of(playerOne, playerTwo) : [playerOne, playerTwo];
    const match = {id, players};
    return immutable ? Map(match) : match;
}