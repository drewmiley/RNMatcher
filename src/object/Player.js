import {Map} from 'immutable';

export default Player = (id, name, rating, map) => {
    const player = {id, name, rating};
    return map ? Map(player) : player;
}