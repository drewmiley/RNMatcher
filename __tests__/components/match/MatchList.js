import 'react-native';
import { List } from 'immutable';
import React from 'react';
import renderer from 'react-test-renderer';

import MatchList from '../../../src/components/match/MatchList';

import Match from '../../../src/object/Match';
import Player from '../../../src/object/Player';

describe('MatchList', () => {

    describe('Rendering', () => {

        it('should match the snapshot', () => {
            const matches = List.of(
                new Match(1, new Player(1, 'Drew', 10, true), new Player(2, 'James', 20, true), true),
                new Match(2, new Player(3, 'Miley', 30, true), new Player(4, 'Test', 40, true), true)
            );
            const component = renderer.create(
                <MatchList matches={matches} />
            );            
            expect(component).toMatchSnapshot();
        });

    });

});