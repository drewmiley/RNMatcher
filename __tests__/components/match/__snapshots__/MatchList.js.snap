exports[`MatchList Rendering should match the snapshot 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Drew
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
       Versus 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      James
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Miley
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
       Versus 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Test
    </Text>
  </View>
</View>
`;
